lib/modules/dosomething/dosomething_signup/includes/Signup.php

```diff
@@ -89,14 +89,23 @@ private function build($data) {
     $this->id = $data->sid;
     $this->created_at = $data->timestamp;
 
-    $user = user_load($data->uid);
+    $drupal_user = user_load($data->uid);
+
+    if (!isset($data->field_northstar_id_value) || $data->field_northstar_id_value === 'NONE') {
+      $northstar_response = dosomething_northstar_get_northstar_user($data->uid);
+      $northstar_response = json_decode($northstar_response);
+
+      if (!empty($northstar_response->data) && !isset($northstar_response->error)) {
+        $northstar_user = $northstar_response->data;
+      }
+    }
 
     $this->user = [
-      'id' => $data->field_northstar_id_value === 'NONE' ? NULL : $data->field_northstar_id_value,
-      'first_name' => dosomething_helpers_extract_field_data($user->field_first_name),
-      'last_initial' => dosomething_helpers_extract_field_data($user->field_last_name),
-      'photo' => $user->picture,
-      'country' => dosomething_helpers_extract_field_data($user->field_address),
+      'id' => isset($northstar_user) ? $northstar_user->id : $data->field_northstar_id_value,
+      'first_name' => dosomething_helpers_extract_field_data($drupal_user->field_first_name),
+      'last_initial' => dosomething_helpers_extract_field_data($drupal_user->field_last_name),
+      'photo' => dosomething_helpers_extract_field_data($drupal_user->photo),
```

Comment:
Wait, do we store the same `photo` on Drupal profiles? I thought that was just for MCC spotlights, rather than the avatar that is normally included in the Northstar response.